kernel/source/messaging/ipc.c

```diff
@@ -67,172 +71,414 @@ Copyright (C) 2009		John Kelley <wiidev@kelley.ca>
 
 #define IPC_MAX_FILENAME	0x1300
 
-extern char __mem2_area_start[];
-static volatile IpcMessage* input_queue[IPC_IN_SIZE] ALIGNED(0x20) SRAM_BSS;
-static volatile IpcMessage* output_queue[IPC_OUT_SIZE] ALIGNED(0x20) SRAM_BSS;
-static u16 in_cnt = 0;
-static u16 out_cnt = 0;
+#define MAX_IPCMESSAGES (MAX_THREADS + 128)
 
-void ipc_send_ack(void)
-{
-	u32 ppc_regs = read32(HW_IPC_PPCCTRL);
-	u32 arm_regs = read32(HW_IPC_ARMCTRL);
-	
-	//Send ACK to the PPC + remove the reply flag
-	ppc_regs &= ~(IPC_TRIG_OUTGOING);
-	ppc_regs |= IPC_TRIG_ACK | IPC_PPC_ACK;
-	arm_regs |= IPC_ARM_ACK_OUT;
-	
-	write32(HW_IPC_PPCCTRL, ppc_regs );
-	write32(HW_IPC_ARMCTRL, arm_regs );
-}
+IpcMessage* IpcMessageArray = NULL;
+FileDescriptorPath* FiledescPathArray = NULL;
+MessageQueue IpcMessageQueueArray[MAX_THREADS] SRAM_BSS;
+unsigned ThreadMessageUsageArray[MAX_THREADS] SRAM_BSS;
+static void* IpcMessageQueueDataPtrArray[MAX_THREADS] SRAM_BSS;
+static FileDescriptorPath* FiledescPathPointerArray[MAX_THREADS] SRAM_BSS;
+
+ThreadInfo* IpcHandlerThread = NULL;
+s32 IpcHandlerThreadId = -1;
+
+static IpcMessage* IpcHandlerMessageQueueData[50] SRAM_BSS;
+static IpcRequest IpcHandlerRequest SRAM_BSS;
```

Comment:
Seems to be already covered by [this line in the thread function](https://github.com/DacoTaco/StarStruck/pull/21/files#diff-aa1ba418678776b708435212e63e873e8b801f71f4164ae60e623c306f00556cR162) ?

Reply:
it is, but i'd consider it cleaner to have it init as it is, instead of the thread doing it. i know its not how ios does it, but the result should be the same right? (besides the variable no longer being in bss)